src/com/etendoerp/metadata/MetadataServlet.java

```diff
@@ -42,9 +43,84 @@ private void handleRequest(HttpServletRequest request, HttpServletResponse respo
             response.getWriter().write(this.fetchMenu().toString());
         } else if (path.equals("/session")) {
             response.getWriter().write(this.fetchSession().toString());
+        } else if (path.startsWith("/toolbar")) {
+            String[] pathParts = path.split("/");
+            if (pathParts.length < 3) {
+                sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, "Invalid toolbar path");
+                return;
+            }
+            try {
+                String windowId = pathParts[2];
+                String tabId = (pathParts.length >= 4 && !"undefined".equals(pathParts[3])) ? pathParts[3] : null;
+                JSONObject toolbar = fetchToolbar(windowId, tabId);
+                sendSuccessResponse(response, toolbar);
+            } catch (IllegalArgumentException e) {
+                sendErrorResponse(response, HttpServletResponse.SC_NOT_FOUND, e.getMessage());
```

Comment:
Please, create a new exception called `UnprocessableContentException`.
Remove this sendErrorResponse and instead use:

`throw new UnprocessableContentException(e.getMessage());`

you will need to update the `getResponseStatus` of `BaseServlet`:


```
    protected int getResponseStatus(Exception e) {
        Object clazz = e.getClass();

        if (clazz.equals(OBSecurityException.class) || clazz.equals(UnauthorizedException.class)) return 401;
        else if (clazz.equals(MethodNotAllowedException.class)) return 405;
        else if (clazz.equals(UnprocessableContentException.class)) return 422;
        else return 500;
    }
```